core/src/main/java/io/substrait/function/ToTypeString.java

```diff
@@ -82,12 +82,12 @@ public String visit(final Type.Timestamp expr) {
 
   @Override
   public String visit(final Type.IntervalYear expr) {
-    return "year";
+    return "iyear";
   }
 
   @Override
   public String visit(final Type.IntervalDay expr) {
-    return "day";
+    return "iday";
   }
```

Comment:
These now match what is expected based on https://substrait.io/extensions/#function-signature-compound-names